drivers/at86rf215/at86rf215_netdev.c

```diff
@@ -0,0 +1,922 @@
+            res = sizeof(netopt_enable_t);
+            break;
+
+        case NETOPT_CSMA_RETRIES:
+            assert(len <= sizeof(uint8_t));
+            dev->csma_retries_max = *((const uint8_t *)val);
+            res = sizeof(uint8_t);
+            break;
+
+        case NETOPT_CSMA_MAXBE:
+            assert(len <= sizeof(uint8_t));
+            dev->csma_maxbe = *((const uint8_t *)val);
+            res = sizeof(uint8_t);
+            break;
+
+        case NETOPT_CSMA_MINBE:
+            assert(len <= sizeof(uint8_t));
+            dev->csma_minbe = *((const uint8_t *)val);
+            res = sizeof(uint8_t);
+            break;
+
+        case NETOPT_CCA_THRESHOLD:
+            assert(len <= sizeof(int8_t));
+            at86rf215_set_cca_threshold(dev, *((const int8_t *)val));
+            res = sizeof(int8_t);
+            break;
+
+        default:
+            break;
+    }
+
+    if (res == -ENOTSUP) {
+        res = netdev_ieee802154_set((netdev_ieee802154_t *)netdev, opt, val, len);
+    }
+
+    return res;
+}
+
+static void _enable_tx2rx(at86rf215_t *dev)
+{
+    uint8_t amcs = at86rf215_reg_read(dev, dev->BBC->RG_AMCS);
+
+    /* disable AACK, enable TX2RX */
+    amcs |=  AMCS_TX2RX_MASK;
+    amcs &= ~AMCS_AACK_MASK;
+
+    at86rf215_reg_write(dev, dev->BBC->RG_AMCS, amcs);
+}
+
+static void _tx_end(at86rf215_t *dev, netdev_event_t event)
+{
+    netdev_t *netdev = (netdev_t *)dev;
+
+    /* listen to non-ACK packets again */
+    if (dev->flags & AT86RF215_OPT_ACK_REQUESTED) {
+        dev->flags &= ~AT86RF215_OPT_ACK_REQUESTED;
+        at86rf215_filter_ack(dev, false);
+    }
+
+    at86rf215_tx_done(dev);
+
+    if (dev->flags & AT86RF215_OPT_TELL_TX_END) {
+        netdev->event_callback(netdev, event);
+    }
+
+    dev->timeout = 0;
+    dev->state = AT86RF215_STATE_IDLE;
+}
+
+static void _ack_timeout_cb(void* arg) {
+    at86rf215_t *dev = arg;
+    dev->timeout = AT86RF215_TIMEOUT_ACK;
+    msg_send_int(&dev->timer_msg, dev->timer_msg.sender_pid);
+}
+
+static void _backoff_timeout_cb(void* arg) {
+    at86rf215_t *dev = arg;
+    dev->timeout = AT86RF215_TIMEOUT_CSMA;
+    msg_send_int(&dev->timer_msg, dev->timer_msg.sender_pid);
+}
+
+static void _set_idle(at86rf215_t *dev)
+{
+    dev->state = AT86RF215_STATE_IDLE;
+
+    uint8_t next_state;
+
+    if (dev->flags & AT86RF215_OPT_TX_PENDING) {
+        next_state = CMD_RF_TXPREP;
+    } else {
+        next_state = CMD_RF_RX;
+    }
+
+    at86rf215_rf_cmd(dev, next_state);
+}
+
+/* wake up the radio thread after ACK timeout */
+static void _start_ack_timer(at86rf215_t *dev)
+{
+    dev->timer_msg.type = NETDEV_MSG_TYPE_EVENT;
+    dev->timer_msg.sender_pid = thread_getpid();
+
+    dev->timer.arg = dev;
+    dev->timer.callback = _ack_timeout_cb;
+
+    xtimer_set(&dev->timer, dev->ack_timeout_usec);
+}
+
+/* wake up the radio thread after CSMA backoff period */
+static void _start_backoff_timer(at86rf215_t *dev)
+{
+    uint8_t be;
```

Comment:
whats `be` supposed to mean?

Reply:
backoff exponent probably - comment added.